src/test/soundsourceproviderregistrytest.cpp

```diff
@@ -0,0 +1,164 @@
+#include <QtDebug>
+
+#include <gtest/gtest.h>
+
+#include "sources/soundsourceproviderregistry.h"
+
+namespace Mixxx {
+
+namespace {
+
+class TestSoundSourceProvider: public SoundSourceProvider {
+public:
+    TestSoundSourceProvider(
+            QString name,
+            QStringList supportedFileExtensions,
+            SoundSourceProviderPriority priorityHint)
+            : m_name(name),
+              m_supportedFileExtensions(supportedFileExtensions),
+              m_priorityHint(priorityHint) {
+    }
+
+    QString getName() const override {
+        return m_name;
+    }
+
+    // A list of supported file extensions in any order.
+    QStringList getSupportedFileExtensions() const override {
+        return m_supportedFileExtensions;
+    }
+
+    SoundSourceProviderPriority getPriorityHint(
+            const QString& supportedFileExtension) const override {
+        Q_UNUSED(supportedFileExtension);
+        return m_priorityHint;
+    }
+
+    SoundSourcePointer newSoundSource(const QUrl& /*url*/) override {
+        return SoundSourcePointer();
+    }
+
+private:
+    const QString m_name;
+    const QStringList m_supportedFileExtensions;
+    const SoundSourceProviderPriority m_priorityHint;
+};
+
+class SoundSourceProviderRegistryTest : public testing::Test {
+  protected:
+    SoundSourceProviderRegistryTest() {
+    }
+
+    static const QString FILE_EXT1;
+    static const QString FILE_EXT2;
+
+    void SetUp() override {
+        m_supportedFileExtensions.append(FILE_EXT1);
+        m_supportedFileExtensions.append(FILE_EXT2);
+    }
+
+    void TearDown() override {
+    }
+
+    SoundSourceProviderPointer createProvider(
+            QString name,
+            QStringList supportedFileExtensions,
+            SoundSourceProviderPriority priorityHint = SoundSourceProviderPriority::DEFAULT) {
+        return SoundSourceProviderPointer(
+                new TestSoundSourceProvider(
+                        name, supportedFileExtensions, priorityHint));
+    }
+
+    SoundSourceProviderPointer createProvider(
+            QString name,
+            SoundSourceProviderPriority priorityHint = SoundSourceProviderPriority::DEFAULT) {
+        return SoundSourceProviderPointer(
+                new TestSoundSourceProvider(
+                        name, m_supportedFileExtensions, priorityHint));
+    }
+
+    static QStringList getAllRegisteredProviderNamesForFileExtension(
+            const SoundSourceProviderRegistry& cut, QString fileExt) {
```

Comment:
"class under test" or "component under test", a common naming convention for 
unit tests:

http://c2.com/cgi/wiki?ClassUnderTest

On 11/25/2015 11:24 AM, Daniel Schürmann wrote:
>
> In src/test/soundsourceproviderregistrytest.cpp 
> <https://github.com/mixxxdj/mixxx/pull/605#discussion_r45850562>:
>
> > +            SoundSourceProviderPriority priorityHint = SoundSourceProviderPriority::DEFAULT) {
> > +        return SoundSourceProviderPointer(
> > +                new TestSoundSourceProvider(
> > +                        name, supportedFileExtensions, priorityHint));
> > +    }
> > +
> > +    SoundSourceProviderPointer createProvider(
> > +            QString name,
> > +            SoundSourceProviderPriority priorityHint = SoundSourceProviderPriority::DEFAULT) {
> > +        return SoundSourceProviderPointer(
> > +                new TestSoundSourceProvider(
> > +                        name, m_supportedFileExtensions, priorityHint));
> > +    }
> > +
> > +    static QStringList getAllRegisteredProviderNamesForFileExtension(
> > +            const SoundSourceProviderRegistry& cut, QString fileExt) {
>
> What stands cut for?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/605/files#r45850562>.
>